### ISSUE TITLE
looks confusing in IntelliJ IDEA

### ISSUE THREAD

**Issue by mb73** [agido/pageobject#3]
Running PageObject tests in IntelliJ IDEA looks confusing, because in the tree view, browsers are nested in each other (InternetExplorer in Chrome in Firefox). Is there any chance to get this fixed?

![image](https://cloud.githubusercontent.com/assets/22412208/20008135/a777ce7e-a29f-11e6-8b8e-e707e1eac503.png)

**Comment by drieks** [comment]
This seems to be a bug in IntelliJ related to Multithreaded Test execution. Can you try to use only one Thread?

Add TEST_LIMIT=1 to the environment vars of the launch config to try it (have a look at the checked in launch config "All Tests".)

I'm using only one browser while developing in IDEA to prevent this :disappointed:

**Comment by mb73** [comment]
Thanks, but that does not seem to change anything. Any idea?

![image](https://cloud.githubusercontent.com/assets/22412208/20261500/575f4fb8-aa5e-11e6-937a-06a2328aeb4f.png)

**Comment by drieks** [comment]
ok, it seems there is also another issue.

please try to run only one browser, maybe add
```
HTMLUNIT_LIMIT=0
FIREFOX_LIMIT=0
```
if you are testing with chrome. does this help?

**Comment by mb73** [comment]
Well, somewhat. There is still the Internet Explorer, and `IE_LIMIT=0` does not seem to help. Strangely removing the webdriver.ie.driver VM parameter does not help either, although [in the doc it says IE_LIMIT defaults to 0 in this case](https://github.com/agido/pageobject/blob/648be940bef177edd9b6fe459be595121a5655c9/core/src/main/scala/org/pageobject/core/tools/Limit.scala). Well, at least no IE pops up. But even without any drivers I have still Chrome and InternetExplorer in the tree.

![image](https://cloud.githubusercontent.com/assets/22412208/20479447/80a3338c-afde-11e6-82f4-c6ffc580c30b.png)

**Comment by drieks** [comment]
Hi, are you using version 0.1.0 or 0.2.0-snapshot? I have changed this, the old behavior was to start all possible browsers. But because some browsers can only be started when `-Dwebdriver.xyz.driver` is provided I changed this that only the browsers are started where a driver is provided by default.
Because you are not providing any VM parameters I think you are using version 0.1.0, right?

But this are only defaults, because you are overriding `*_LIMIT` values as we can see in the screenshot this should make no difference to you. I think you have read the documentation online, in this case it may not apply to the 0.1.0 version. I will try to release version 0.2.0 this week.

Could you you please try INTERNETEXPLORER_LIMIT=0 instead of IE_LIMIT=0? I think this should fix this issue. (INTERNETEXPLORER_LIMIT was renamed to IE_LIMIT in 0.2.0 because it is too long..)

**Comment by mb73** [comment]
Yes, 0.1.0. INTERNETEXPLORER_LIMIT=0 helps. Thank you!

**Comment by mb73** [comment]
Yes, 0.1.0. INTERNETEXPLORER_LIMIT=0 helps. Thank you! 

However, it still looks somewhat strange in my case. I have a FeatureSpec with two scenarios in one feature. 

![image](https://cloud.githubusercontent.com/assets/22412208/20564982/edeb1934-b18f-11e6-9723-c38757f1a2fc.png)

In the tree view the second scenario does not as being part of the feature.

![image](https://cloud.githubusercontent.com/assets/22412208/20564958/cdf24792-b18f-11e6-8278-0680da2ab7dd.png)

**Comment by drieks** [comment]
ok, I think you should remove the parallel execution trait. Just replace `PageObjectSuite` with `PageBrowser with DriverLauncher with DefaultDriverProvider`.

**Comment by drieks** [comment]
I'm not sure if the bug is located in scalatest or in idea. running the tests using sbt works fine. until now we used only one browser in idea as a workaround, but we are only using FunSpec.

Another problem is that the scalatest people seems to have no time to response to PRs or bug reports.

I will have a look at this, but I think we have to remove the parallel trait from PageObjectSuite so a developer can mix it in if he want to use it.

**Comment by mb73** [comment]
Great! Replacing PageObjectSuite helps – even with multiple browsers, if I leave HTMLUNIT_LIMIT=0. 

![image](https://cloud.githubusercontent.com/assets/22412208/20591851/e971cde4-b22a-11e6-955f-49c3610d7691.png)

For HTMLUNIT_LIMIT=1, everything is nested in HtmlUnit.

![image](https://cloud.githubusercontent.com/assets/22412208/20591922/48dd21fc-b22b-11e6-85c3-43249bc6bf6f.png)

However, currently HtmlUnit does not work anyway for me (no output, no result) – even with all the browsers disabled.

![image](https://cloud.githubusercontent.com/assets/22412208/20592013/c9e54900-b22b-11e6-862a-9a88c6eed767.png)

So I just leave HTMLUNIT_LIMIT=0.

Removing the parallel trait form PageObjectSuite sounds like a good idea.

As for ScalaTest: I see that you are a [member](https://github.com/scalatest/scalatest/network/members). Can't you commit to their trunk? (I’m sorry if that’s a naive question. I’m new at github.)

**Comment by drieks** [comment]
Hi,

if you click on "fork" you are automatically a member of the network. I have created some Pull Requests and Issues, but I got no responses to PRs..

https://github.com/scalatest/scalatest/issues?utf8=%E2%9C%93&q=author%3Adrieks%20

There is another issue with ParallelTestExecution (scalatest/scalatest#898) but because even this is not solved I don't filled another issue for this problem here.

Because we need a small fix to @WrapWith to run the tests with configurable browsers we need a patched version of scalatest. I hope this will find its way into upstream because others can use this feature as well, it is not directly related to pageobject. We have to wait...

**Comment by drieks** [comment]
There was another place using ParallelTestExecution (in DriverLaunchWrapper) that was causing the unit tests to fail, also affecting test execution in maven. I removed it in commit 02fedd641db630445cfb7b270dd7b2d82a3fd3e5

Sadly it seems that it is not reliable possible to run scalatest tests in parallel. Because ParallelTestExecution is a scalatest trait we can't really do anything about this.

Running in parallel will cause:
* idea will show confusing test results
* tests are executed in another thread when using maven. but the main thread is not waiting for the tests and will just continue. I will tell something like "I'm finished!" and just terminate the VM while the tests are still running...


Summary: don't use ParallelTestExecution unless you really have to do so...

**Comment by mb73** [comment]
Looks fine on PageObject 0.2.0 (snapshot). Thank you!
